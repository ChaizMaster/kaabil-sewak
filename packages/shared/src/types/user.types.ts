export interface User {
  id: string;
  name: string;
  phone: string;
  email?: string;
  preferredLanguage: Language;
  location: UserLocation;
  skills: string[];
  profileComplete: boolean;
  createdAt: string;
  updatedAt: string;
}

export enum Language {
  ENGLISH = 'en',
  HINDI = 'hi',
  BENGALI = 'bn'
}

export interface UserLocation {
  address: string;
  city: string;
  state: string;
  pincode: string;
  coordinates?: {
    latitude: number;
    longitude: number;
  };
}

export interface AuthState {
  isAuthenticated: boolean;
  user: User | null;
  onboardingComplete: boolean;
  languageSelected: boolean;
}

export interface SignUpData {
  name: string;
  phone: string;
  email?: string;
  preferredLanguage: Language;
} 