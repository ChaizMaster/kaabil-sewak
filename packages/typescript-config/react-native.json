{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "React Native",
  "extends": "./base.json",
  "compilerOptions": {
    "jsx": "react-native",
    "lib": ["es2022", "DOM"],
    "target": "es2022"
  }
} 