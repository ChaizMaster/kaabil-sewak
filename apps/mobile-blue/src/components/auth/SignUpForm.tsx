import React, { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, StyleSheet, Alert } from 'react-native';
import { Language, SignUpData } from 'shared/src/types/user.types';
import { useTranslation } from 'shared/src/hooks/useTranslation';

interface SignUpFormProps {
  onSignUp: (data: SignUpData) => void;
  onSwitchToLogin: () => void;
  language: Language;
  isLoading?: boolean;
  error?: string;
}

interface FormErrors {
  name?: string;
  phone?: string;
  email?: string;
}

export const SignUpForm: React.FC<SignUpFormProps> = ({
  onSignUp,
  onSwitchToLogin,
  language,
  isLoading = false,
  error,
}) => {
  const { t } = useTranslation(language);
  const [formData, setFormData] = useState({
    name: '',
    phone: '',
    email: '',
  });
  const [errors, setErrors] = useState<FormErrors>({});

  const validateForm = (): boolean => {
    const newErrors: FormErrors = {};

    if (!formData.name.trim()) {
      newErrors.name = 'Name is required';
    }

    if (!formData.phone.trim()) {
      newErrors.phone = 'Phone number is required';
    } else if (!/^\d{10}$/.test(formData.phone.trim())) {
      newErrors.phone = 'Please enter a valid 10-digit phone number';
    }

    setErrors(newErrors);
    return Object.keys(newErrors).length === 0;
  };

  const handleSubmit = () => {
    if (validateForm()) {
      const signUpData: SignUpData = {
        name: formData.name.trim(),
        phone: formData.phone.trim(),
        email: formData.email.trim() || undefined,
        preferredLanguage: language,
      };
      onSignUp(signUpData);
    }
  };

  const updateField = (field: keyof typeof formData, value: string) => {
    setFormData(prev => ({ ...prev, [field]: value }));
    // Clear error when user starts typing
    if (errors[field]) {
      setErrors(prev => ({ ...prev, [field]: undefined }));
    }
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{t.signUp}</Text>

      {error && (
        <View style={styles.errorContainer}>
          <Text style={styles.errorText}>{error}</Text>
        </View>
      )}

      <View style={styles.formContainer}>
        <View style={styles.inputGroup}>
          <Text style={styles.label}>{t.name}</Text>
          <TextInput
            testID="name-input"
            accessibilityLabel="Enter your full name"
            style={[styles.input, errors.name && styles.inputError]}
            value={formData.name}
            onChangeText={(text) => updateField('name', text)}
            placeholder={t.name}
            placeholderTextColor="#999"
            autoCapitalize="words"
            autoComplete="name"
          />
          {errors.name && <Text style={styles.fieldError}>{errors.name}</Text>}
        </View>

        <View style={styles.inputGroup}>
          <Text style={styles.label}>{t.phone}</Text>
          <TextInput
            testID="phone-input"
            accessibilityLabel="Enter your phone number"
            style={[styles.input, errors.phone && styles.inputError]}
            value={formData.phone}
            onChangeText={(text) => updateField('phone', text)}
            placeholder="9876543210"
            placeholderTextColor="#999"
            keyboardType="phone-pad"
            autoComplete="tel"
            maxLength={10}
          />
          {errors.phone && <Text style={styles.fieldError}>{errors.phone}</Text>}
        </View>

        <View style={styles.inputGroup}>
          <Text style={styles.label}>{t.email}</Text>
          <TextInput
            testID="email-input"
            accessibilityLabel="Enter your email address (optional)"
            style={[styles.input, errors.email && styles.inputError]}
            value={formData.email}
            onChangeText={(text) => updateField('email', text)}
            placeholder="example@gmail.com"
            placeholderTextColor="#999"
            keyboardType="email-address"
            autoCapitalize="none"
            autoComplete="email"
          />
          {errors.email && <Text style={styles.fieldError}>{errors.email}</Text>}
        </View>

        <TouchableOpacity
          testID="signup-button"
          style={[styles.signUpButton, isLoading && styles.disabledButton]}
          onPress={handleSubmit}
          disabled={isLoading}
        >
          <Text style={styles.signUpButtonText}>
            {isLoading ? t.loading : t.createAccount}
          </Text>
        </TouchableOpacity>

        <TouchableOpacity
          testID="login-link"
          style={styles.loginLink}
          onPress={onSwitchToLogin}
        >
          <Text style={styles.loginLinkText}>{t.alreadyHaveAccount}</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: '#fff',
    justifyContent: 'center',
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
    textAlign: 'center',
    marginBottom: 30,
    color: '#2E86AB',
  },
  errorContainer: {
    backgroundColor: '#FFE6E6',
    padding: 12,
    borderRadius: 8,
    marginBottom: 20,
  },
  errorText: {
    color: '#D32F2F',
    textAlign: 'center',
    fontSize: 14,
  },
  formContainer: {
    gap: 20,
  },
  inputGroup: {
    gap: 8,
  },
  label: {
    fontSize: 16,
    fontWeight: '600',
    color: '#333',
  },
  input: {
    borderWidth: 2,
    borderColor: '#E0E0E0',
    borderRadius: 12,
    padding: 16,
    fontSize: 16,
    backgroundColor: '#F8F9FA',
    minHeight: 44, // Accessibility: minimum touch target
  },
  inputError: {
    borderColor: '#D32F2F',
  },
  fieldError: {
    color: '#D32F2F',
    fontSize: 12,
    marginTop: 4,
  },
  signUpButton: {
    backgroundColor: '#2E86AB',
    padding: 16,
    borderRadius: 12,
    alignItems: 'center',
    marginTop: 10,
    minHeight: 44, // Accessibility: minimum touch target
  },
  disabledButton: {
    backgroundColor: '#CCC',
  },
  signUpButtonText: {
    color: '#fff',
    fontSize: 18,
    fontWeight: '600',
  },
  loginLink: {
    alignItems: 'center',
    marginTop: 20,
    padding: 10,
  },
  loginLinkText: {
    color: '#2E86AB',
    fontSize: 16,
    textDecorationLine: 'underline',
  },
}); 