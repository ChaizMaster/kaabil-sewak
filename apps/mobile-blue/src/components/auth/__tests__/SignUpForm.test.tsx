import React from 'react';
import { render, fireEvent, screen, waitFor } from '@testing-library/react-native';
import { SignUpForm } from '../SignUpForm';
import { Language } from 'shared/src/types/user.types';

// Mock the useTranslation hook
jest.mock('shared/src/hooks/useTranslation', () => ({
  useTranslation: (lang: Language) => ({
    t: {
      signUp: 'Sign Up',
      name: 'Full Name',
      phone: 'Phone Number',
      email: 'Email (Optional)',
      photo: 'Photo',
      photoRequired: 'Photo is required',
      photoIdentityCaption: 'Please upload your photo or take a selfie to uniquely identify you, as names can be common. This helps us ensure you are a genuine worker.',
      uploadPhoto: 'Upload Photo',
      takePhoto: 'Take Photo',
      chooseFromGallery: 'Choose from Gallery',
      retakePhoto: 'Retake Photo',
      cancel: 'Cancel',
      createAccount: 'Create Account',
      alreadyHaveAccount: 'Already have an account? Login',
      loading: 'Loading...',
      error: 'Something went wrong',
    },
    currentLanguage: lang,
  }),
}));

// Mock the LanguageChangeModal
jest.mock('../../common/LanguageChangeModal', () => ({
  LanguageChangeModal: ({ visible, onClose, onLanguageChange }: any) => {
    const { Text, TouchableOpacity, View } = require('react-native');
    return visible ? (
      <View testID="language-modal">
        <Text>Language Modal</Text>
        <TouchableOpacity
          testID="select-hindi"
          onPress={() => {
            onLanguageChange(Language.HINDI);
            onClose();
          }}
        >
          <Text>Select Hindi</Text>
        </TouchableOpacity>
        <TouchableOpacity testID="close-modal" onPress={onClose}>
          <Text>Close</Text>
        </TouchableOpacity>
      </View>
    ) : null;
  },
}));

// Mock the PhotoUpload component
jest.mock('../../common/PhotoUpload', () => ({
  PhotoUpload: ({ onPhotoSelected, error }: any) => {
    const { Text, TouchableOpacity, View } = require('react-native');
    return (
      <View testID="photo-upload">
        <Text>Photo Upload Component</Text>
        <TouchableOpacity
          testID="mock-photo-select"
          onPress={() => onPhotoSelected('mock-photo-uri')}
        >
          <Text>Select Photo</Text>
        </TouchableOpacity>
        {error && <Text testID="photo-error">{error}</Text>}
      </View>
    );
  },
}));

describe('SignUpForm Component', () => {
  const mockOnSignUp = jest.fn();
  const mockOnSwitchToLogin = jest.fn();
  const mockOnBack = jest.fn();
  const mockOnLanguageChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders all form fields', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    expect(screen.getByTestId('name-input')).toBeTruthy();
    expect(screen.getByTestId('phone-input')).toBeTruthy();
    expect(screen.getByTestId('email-input')).toBeTruthy();
    expect(screen.getByTestId('signup-button')).toBeTruthy();
  });

  test('displays form labels in correct language', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    expect(screen.getByText('Full Name')).toBeTruthy();
    expect(screen.getByText('Phone Number')).toBeTruthy();
    expect(screen.getByText('Email (Optional)')).toBeTruthy();
  });

  test('validates required fields', async () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    const signUpButton = screen.getByTestId('signup-button');
    fireEvent.press(signUpButton);
    
    await waitFor(() => {
      expect(screen.getByText('Name is required')).toBeTruthy();
      expect(screen.getByText('Phone number is required')).toBeTruthy();
      expect(screen.getByText('Photo is required')).toBeTruthy();
    });
  });

  test('validates phone number format', async () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    const phoneInput = screen.getByTestId('phone-input');
    fireEvent.changeText(phoneInput, '123');
    
    const signUpButton = screen.getByTestId('signup-button');
    fireEvent.press(signUpButton);
    
    await waitFor(() => {
      expect(screen.getByText('Please enter a valid 10-digit phone number')).toBeTruthy();
    });
  });

  test('calls onSignUp with correct data when form is valid', async () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    const nameInput = screen.getByTestId('name-input');
    const phoneInput = screen.getByTestId('phone-input');
    const emailInput = screen.getByTestId('email-input');
    const photoSelectButton = screen.getByTestId('mock-photo-select');
    
    fireEvent.changeText(nameInput, 'Ravi Kumar');
    fireEvent.changeText(phoneInput, '9876543210');
    fireEvent.changeText(emailInput, 'ravi@example.com');
    fireEvent.press(photoSelectButton);
    
    const signUpButton = screen.getByTestId('signup-button');
    fireEvent.press(signUpButton);
    
    await waitFor(() => {
      expect(mockOnSignUp).toHaveBeenCalledWith({
        name: 'Ravi Kumar',
        phone: '9876543210',
        email: 'ravi@example.com',
        photo: 'mock-photo-uri',
        preferredLanguage: Language.ENGLISH,
      });
    });
  });

  test('allows signup without email', async () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    const nameInput = screen.getByTestId('name-input');
    const phoneInput = screen.getByTestId('phone-input');
    const photoSelectButton = screen.getByTestId('mock-photo-select');
    
    fireEvent.changeText(nameInput, 'Raj Singh');
    fireEvent.changeText(phoneInput, '9123456789');
    fireEvent.press(photoSelectButton);
    
    const signUpButton = screen.getByTestId('signup-button');
    fireEvent.press(signUpButton);
    
    await waitFor(() => {
      expect(mockOnSignUp).toHaveBeenCalledWith({
        name: 'Raj Singh',
        phone: '9123456789',
        email: undefined,
        photo: 'mock-photo-uri',
        preferredLanguage: Language.ENGLISH,
      });
    });
  });

  test('shows loading state during submission', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
        isLoading={true}
      />
    );
    
    expect(screen.getByText('Loading...')).toBeTruthy();
    expect(screen.getByTestId('signup-button')).toBeDisabled();
  });

  test('displays error message when provided', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
        error="Phone number already exists"
      />
    );
    
    expect(screen.getByText('Phone number already exists')).toBeTruthy();
  });

  test('calls onSwitchToLogin when login link is pressed', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    const loginLink = screen.getByTestId('login-link');
    fireEvent.press(loginLink);
    
    expect(mockOnSwitchToLogin).toHaveBeenCalled();
  });

  test('has proper accessibility labels', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    expect(screen.getByLabelText('Enter your full name')).toBeTruthy();
    expect(screen.getByLabelText('Enter your phone number')).toBeTruthy();
    expect(screen.getByLabelText('Enter your email address (optional)')).toBeTruthy();
  });

  test('shows back button when onBack prop is provided', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    expect(screen.getByTestId('back-button')).toBeTruthy();
  });

  test('hides back button when onBack prop is not provided', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={undefined}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    expect(screen.queryByTestId('back-button')).toBeNull();
  });

  test('calls onBack when back button is pressed', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    const backButton = screen.getByTestId('back-button');
    fireEvent.press(backButton);
    
    expect(mockOnBack).toHaveBeenCalledTimes(1);
  });

  test('shows language indicator with correct flag and text', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    expect(screen.getByTestId('language-change-button')).toBeTruthy();
    expect(screen.getByText('🇺🇸 EN')).toBeTruthy();
  });

  test('shows correct language indicator for Hindi', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.HINDI}
      />
    );
    expect(screen.getByText('🇮🇳 हि')).toBeTruthy();
  });

  test('shows correct language indicator for Bengali', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.BENGALI}
      />
    );
    expect(screen.getByText('🇮🇳 বা')).toBeTruthy();
  });

  test('opens language modal when language indicator is pressed', async () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    // Modal should not be visible initially
    expect(screen.queryByTestId('language-modal')).toBeNull();
    
    // Press language indicator
    const languageChangeButton = screen.getByTestId('language-change-button');
    fireEvent.press(languageChangeButton);
    
    // Modal should be visible
    await waitFor(() => {
      expect(screen.queryByTestId('language-modal')).toBeTruthy();
    });
  });

  test('handles language change through modal', async () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    // Open modal
    const languageChangeButton = screen.getByTestId('language-change-button');
    fireEvent.press(languageChangeButton);
    
    // Select Hindi
    await waitFor(() => {
      expect(screen.queryByTestId('language-modal')).toBeTruthy();
    });
    
    const selectHindiButton = screen.getByTestId('select-hindi');
    fireEvent.press(selectHindiButton);
    
    // Verify onLanguageChange was called
    expect(mockOnLanguageChange).toHaveBeenCalledWith(Language.HINDI);
  });

  test('renders photo upload component', () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    expect(screen.getByTestId('photo-upload')).toBeTruthy();
    expect(screen.getByText('Photo Upload Component')).toBeTruthy();
  });

  test('validates photo is required', async () => {
    render(
      <SignUpForm 
        onSignUp={mockOnSignUp} 
        onSwitchToLogin={mockOnSwitchToLogin}
        onBack={mockOnBack}
        onLanguageChange={mockOnLanguageChange}
        language={Language.ENGLISH}
      />
    );
    
    const nameInput = screen.getByTestId('name-input');
    const phoneInput = screen.getByTestId('phone-input');
    
    fireEvent.changeText(nameInput, 'Test User');
    fireEvent.changeText(phoneInput, '9876543210');
    
    const signUpButton = screen.getByTestId('signup-button');
    fireEvent.press(signUpButton);
    
    await waitFor(() => {
      expect(screen.getByTestId('photo-error')).toBeTruthy();
      expect(screen.getByText('Photo is required')).toBeTruthy();
    });
  });
}); 