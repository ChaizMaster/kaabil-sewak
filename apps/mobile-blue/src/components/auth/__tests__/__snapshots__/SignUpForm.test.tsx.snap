// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginTop": 20,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessibilityLabel="Change language"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#F8F9FA",
          "borderColor": "#E0E0E0",
          "borderRadius": 8,
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 12,
          "paddingVertical": 6,
        }
      }
      testID="language-change-button"
    >
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        ðŸ‡ºðŸ‡¸ EN
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 40,
        "paddingHorizontal": 20,
      }
    }
    keyboardShouldPersistTaps="handled"
    showsVerticalScrollIndicator={false}
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#2E86AB",
            "fontSize": 28,
            "fontWeight": "bold",
            "marginBottom": 30,
            "textAlign": "center",
          }
        }
      >
        Sign Up
      </Text>
      <View
        style={
          {
            "gap": 20,
          }
        }
      >
        <View
          style={
            {
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#333",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Full Name
          </Text>
          <TextInput
            accessibilityLabel="Enter your full name"
            autoCapitalize="words"
            autoComplete="name"
            onChangeText={[Function]}
            placeholder="Full Name"
            placeholderTextColor="#999"
            style={
              [
                {
                  "backgroundColor": "#F8F9FA",
                  "borderColor": "#E0E0E0",
                  "borderRadius": 12,
                  "borderWidth": 2,
                  "fontSize": 16,
                  "minHeight": 44,
                  "padding": 16,
                },
                undefined,
              ]
            }
            testID="name-input"
            value=""
          />
        </View>
        <View
          style={
            {
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#333",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Phone Number
          </Text>
          <TextInput
            accessibilityLabel="Enter your phone number"
            autoComplete="tel"
            keyboardType="phone-pad"
            maxLength={10}
            onChangeText={[Function]}
            placeholder="9876543210"
            placeholderTextColor="#999"
            style={
              [
                {
                  "backgroundColor": "#F8F9FA",
                  "borderColor": "#E0E0E0",
                  "borderRadius": 12,
                  "borderWidth": 2,
                  "fontSize": 16,
                  "minHeight": 44,
                  "padding": 16,
                },
                undefined,
              ]
            }
            testID="phone-input"
            value=""
          />
        </View>
        <View
          style={
            {
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#333",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Photo
             *
          </Text>
          <Text
            style={
              {
                "color": "#666",
                "fontSize": 14,
                "lineHeight": 20,
                "marginBottom": 8,
              }
            }
          >
            Please upload your photo or take a selfie to uniquely identify you, as names can be common. This helps us ensure you are a genuine worker.
          </Text>
          <View
            accessibilityLabel="Upload Photo"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderColor": "#E0E0E0",
                "borderRadius": 12,
                "borderStyle": "dashed",
                "borderWidth": 2,
                "height": 200,
                "opacity": 1,
                "overflow": "hidden",
              }
            }
            testID="photo-upload-button"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "gap": 12,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 48,
                  }
                }
              >
                ðŸ“·
              </Text>
              <Text
                style={
                  {
                    "color": "#666",
                    "fontSize": 16,
                    "fontWeight": "500",
                  }
                }
              >
                Upload Photo
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#2E86AB",
              "borderRadius": 12,
              "marginTop": 10,
              "minHeight": 44,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="signup-button"
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "600",
              }
            }
          >
            Create Account
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "marginTop": 20,
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="login-link"
        >
          <Text
            style={
              {
                "color": "#2E86AB",
                "fontSize": 16,
                "textDecorationLine": "underline",
              }
            }
          >
            Already have an account? Login
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
